lib/workers/repository/update/branch/get-updated.ts

```diff
@@ -526,3 +548,168 @@ async function applyManagerBumpPackageVersion(
 
   return result.bumpedContent;
 }
+
+function updateBranchConfigAfterArtifacts(config: BranchConfig): void {
+  //config.schedule = [];
+  config.commitBody = '[SECURITY] - This update addresses a vulnerability.';
+  config.prCreation = 'immediate';
+  config.prHeader =
+    'This PR was generated to address a detected security vulnerability.';
+}
+
+function isFileAddition(file: unknown): file is FileAddition {
+  return (
+    !!file && typeof file === 'object' && (file as any).type === 'addition'
+  );
+}
+
+function parseRpmtestArtifactsResults(results: UpdateArtifactsResult[] | null):
+  | {
+      name: string;
+      oldVersion: string;
+      newVersion: string;
+    }[]
+  | null {
+  if (!results) {
+    return null;
+  }
+  const rpmRegex = /^(?<name>.+)-(?<version>[^-]+-[^-]+)\.src\.rpm$/;
+  const changes = results.flatMap((res) => {
+    let parsedContents = null;
+    let parsedPreviousContents = null;
+    const rpmVersionChanges: {
+      name: string;
+      oldVersion: string;
+      newVersion: string;
+    }[] = [];
+    if (isFileAddition(res.file)) {
+      try {
+        if (res.file.contents) {
+          parsedContents = yaml.load(res.file.contents.toString());
```

Comment:
I know this is a PoC, but parsing the file can be done using the `zod` library, like in our fork where we already have the schema: https://github.com/redhat-exd-rebuilds/renovate/blob/develop/lib/modules/manager/rpm/schema.ts It could make the code after this a bit easier to read.